src/classes/TemporaryFile.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  S3Client,
+  CreateMultipartUploadCommand,
+  UploadPartCommand,
+  CompleteMultipartUploadCommand,
+} from '@aws-sdk/client-s3';
+import { v4 as uuid } from 'uuid';
+import { SystemConfigurationError } from '../errors';
+import type {
+  CompletedPart,
+} from '@aws-sdk/client-s3';
+
+export class TemporaryFile {
+  public readonly permanentFileSystemPath: string;
+
+  private uploadedLength = 0;
+
+  private s3Url?: string;
+
+  private uploadId?: string;
+
+  private readonly bucket: string;
+
+  private readonly key: string;
+
+  private partNumber = 1;
+
+  private readonly s3: S3Client;
+
+  private readonly completedParts: CompletedPart[] = [];
+
+  public constructor(permanentFileSystemPath: string) {
+    if (process.env.TEMPORARY_FILE_S3_BUCKET === undefined) {
+      throw new SystemConfigurationError('TEMPORARY_FILE_S3_BUCKET must be populated in order to upload to s3.');
+    }
+    if (process.env.AWS_ACCESS_KEY_ID === undefined) {
+      throw new SystemConfigurationError('AWS_ACCESS_KEY_ID must be populated in order to upload to s3.');
+    }
+    if (process.env.AWS_ACCESS_SECRET === undefined) {
+      throw new SystemConfigurationError('AWS_ACCESS_SECRET must be populated in order to upload to s3.');
+    }
+    if (process.env.TEMPORARY_FILE_S3_BUCKET_REGION === undefined) {
+      throw new SystemConfigurationError('TEMPORARY_FILE_S3_BUCKET_REGION must be populated in order to upload to s3.');
+    }
+    const path = `${process.env.TEMPORARY_FILE_S3_SUBDIRECTORY ?? ''}/unprocessed`;
+    this.s3 = new S3Client({
+      region: process.env.TEMPORARY_FILE_S3_BUCKET_REGION,
+      credentials: {
+        accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+        secretAccessKey: process.env.AWS_ACCESS_SECRET,
+      },
+    });
+    this.bucket = process.env.TEMPORARY_FILE_S3_BUCKET;
+    this.key = `${path}/${uuid()}`;
+    this.permanentFileSystemPath = permanentFileSystemPath;
+  }
+
+  public get url(): string {
+    if (this.s3Url === undefined) {
+      throw new Error();
```

Comment:
What is this error?

Reply:
I realized this should not error but should just return an empty string if no URL is set.